internal/querywatcher/query_watcher.go

```diff
@@ -240,15 +289,23 @@ func (w *QueryWatcher) clientCount(clients *sync.Map) int {
 }
 
 // runQuery executes a query on its respective cache.
-func (w *QueryWatcher) runQuery(query *DSQLQuery) (*[]dstore.DSQLQueryResultRow, error) {
+func (w *QWatchManager) runQuery(query *DSQLQuery) (*[]dstore.DSQLQueryResultRow, error) {
 	w.QueryCacheMu.RLock()
 	defer w.QueryCacheMu.RUnlock()
 
-	store, ok := w.QueryCache.Get(query.String())
+	queryFingerprint := generateQueryFingerprint(query)
```

Comment:
We're calling `generateQueryFingerprint` for `addWatcher`, `removeWatcher` and `runquery` in which `runQuery` might be the case where the calls are very frequent.
May be in another PR we can think of a way to cache these fingerprints with `query.toString()` depending on it's impact. Would that be required?

Reply:
Makes sense, have moved it to DSQL such that it's generated once.